### ISSUE TITLE
turn off prelaunch "waiting for..."  I don't want that showing up

### ISSUE THREAD

**Issue by Electro-Jam** [microsoft/vscode#232451]
Type: <b>Bug</b>

I am doing embedded programming and I have a prelaunch task to wait for the debugger to start up.  This seems to trigger a new pop-up that says waiting for prelaunch task... and I know this and I don't want to see this pop-up.  I would like to have it turned off.  When I asked copilot to turn this off it took me to a setting that did not impact the pop-up.  

I would love to figure out how to turn this pop-up off... it is not needed and I don't wish to have it.

Extension version: 0.21.1
VS Code version: Code 1.94.0 (d78a74bcdfad14d5d3b1b782f87255d802b57511, 2024-10-02T13:08:12.626Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 7 Microsoft Surface (R) Edition (16 x 1996)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.45GB (7.88GB free)|
|Process Argv|--crash-reporter-id 114a894a-a9e4-4018-8946-bd9bb51920b9|
|Screen Reader|yes|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805:30301674
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
vscrpc:30673769
962ge761:30959799
pythongtdpath:30769146
pythonnoceb:30805159
asynctok:30898717
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
724cj586:31013169
a69g1124:31058053
dvdeprecation:31068756
dwnewjupyter:31046869
newcmakeconfigv2:31071590
impr_priority:31102340
nativerepl2:31139839
refactort:31108082
pythonrstrctxt:31112756
flightc:31134773
wkspc-onlycs-t:31132770
wkspc-ranged-t:31151552
cf971741:31144450
autoexpandse:31146404
iacca1:31150322
notype1cf:31151524
5fd0e150:31146321

```

</details>

**Comment by roblourens** [comment]
I'm unsure about this because the popup conveys important info that explains why you are waiting for debugging to start, and the task may even be stuck. I am wary of adding a setting so people can hide it, and then being confused. But I see why it could be annoying in your scenario when you know that it will appear, especially if it adds screenreader noise.

@meganrogge any thoughts on this from an accessibility perspective?

**Comment by roblourens** [comment]
Maybe we could let you tune the delay before it appears, so you can expect a longer wait without missing it entirely?

**Comment by meganrogge** [comment]
I have not heard similar feedback. I think it's dangerous to have a setting that disables this behavior. 

Can you figure out why that is always popping up instead?

**Comment by meganrogge** [comment]
I actually see this happening in one of my repos and it is annoying. This happens if the prelaunch task is always active IE is watching files for updates. Maybe a setting is a good idea. Let's get more feedback.

**Comment by roblourens** [comment]
It would happen if the background task is not configured properly and actually blocks debugging, in which case it's important. Or if it's just slow, in which case it's just noise.

**Comment by roblourens** [comment]
Oh funny, we apparently just implemented this setting: https://github.com/microsoft/vscode/issues/228640

You got the update today, try it out :)

**Comment by HarisHashim** [comment]
@roblourens  I tested what mentioned in #228640

Let say if the dialog is displayed and I click debug anyway. Debugging will proceed  without any issue.

The implementation of  #228640 hide the dialog but debugging never proceed to happened. It just stalled.

Perhaps this is another issue. But let me know if this is the expected behavior?

What I expect is debugging continue since clicking debug anyway cause that to happen.

**Comment by ZuBB** [comment]
I have similar behaviour to one that mentioned by @HarisHashim.

When subject option is off -- I click the start button and after some time (my build is not fast) browser window with my app shows up.

When it enabled -- no browser window is shown, does not matter how long I wait (in builtin terminal pane all OK, I see that app has been started). When I became bored -- I press Ctrl+C, app stops in terminal, and in the next moment browser starts with some leftovers of my app.

it looks like this option is better to be turned off

**Comment by ZuBB** [comment]
> I have similar behaviour to one that mentioned by [@HarisHashim](https://github.com/HarisHashim).
> 
> When subject option is off -- I click the start button and after some time (my build is not fast) browser window with my app shows up.
> 
> When it enabled -- no browser window is shown, does not matter how long I wait (in builtin terminal pane all OK, I see that app has been started). When I became bored -- I press Ctrl+C, app stops in terminal, and in the next moment browser starts with some leftovers of my app.
> 
> it looks like this option is better to be turned off

I wanna diregard my comment. I mostly use Chromium for dev stuff, but despite it is an acestor for all Google Chrome family (dev, beta, canary), vscode has issues with using it (path to it in `runtimeExecutable` ) in `launch` request. with `beta` all is ok.
